noir-projects/aztec-nr/aztec/src/state_vars/private_mutable.nr

```diff
@@ -85,6 +85,26 @@ impl<Note> PrivateMutable<Note> {
     }
     // docs:end:replace
 
+    pub fn initialize_or_replace<N>(self, note: &mut Note, broadcast: bool, ivpk_m: GrumpkinPoint) where Note: NoteInterface<N> {
+        let is_initialized = check_nullifier_exists(self.compute_initialization_nullifier());
+
+        // check_nullifier_exists() is an unconstrained function - we can constrain a true value by providing an 
```

Comment:
Quite nice explanation. Might be good to add a note that this is only safe when the note don't influence the initialization nullifier. If it did the same storage var could be initialized multiple times. 

Might be a good idea for us to generally add what is assumed for safety on all of these checks that we have here and there.